google-cloud-storage/src/test/java/com/google/cloud/storage/it/GrpcPlainRequestLoggingInterceptor.java

```diff
@@ -223,6 +245,118 @@ private static ByteString snipBytes(ByteString content) {
     return content.substring(0, 20).concat(snip);
   }
 
+  // while trailers.get can return null, we're always getting values based on the keys it told us
+  // it had.
+  @SuppressWarnings("DataFlowIssue")
+  @VisibleForTesting
+  public static @NonNull Supplier<String> lazyOnCloseLogString(Status status, Metadata trailers) {
+    return () -> {
+      final StringBuilder sb = new StringBuilder();
+      String description = status.getDescription();
+      sb.append("<<< status = {").append("\n  code[4]=").append(status.getCode());
+      if (description != null) {
+        sb.append(",\n  description[")
+            .append(description.getBytes(StandardCharsets.US_ASCII).length)
+            .append("]='")
+            .append(description)
+            .append("'");
+      }
+      sb.append("\n}, \ntrailers = {");
```

Comment:
This logic seems complex enough to deserve some sort of unit test.

Reply:
Test added